### ISSUE TITLE
Auto Scroll to current word

### ISSUE THREAD

**Issue by rogueyoshi** [jamestomasino/stutter#58]
Hey, it would be cool if you had a feature to have the page scroll with and even highlight the current word being shown while reading. It helps a lot to know where you are in a page/article.

**Comment by jamestomasino** [comment]
This is related to #3 and has the same limitations. We're not connecting the stutter content directly with the DOM. It's passed through a readability filter or it's pulling content from the selection text. Without a major rewrite and some other method of parsing these aren't feasible.

**Comment by rogueyoshi** [comment]
I know one easy way to add this possibly. Text Fragment API.

https://web.dev/text-fragments/

**Comment by jamestomasino** [comment]
Looks like that's still in draft and chrome-only at the moment, but you're right. It does represent a possible angle to reconnect the text to the DOM in some way. I'll leave this open until that spec gets a bit further.

**Comment by rodfersou** [comment]
https://openbase.com/js/text-fragments-polyfill/documentation

**Comment by jamestomasino** [comment]
Oooh, thank you. I know what I'm playing with this weekend!

**Comment by rodfersou** [comment]
you are welcome!

**Comment by rodfersou** [comment]
thank you for making my read experience better! 😉

**Comment by jamestomasino** [comment]
Here's a progress update: https://vimeo.com/562015790

This process required me to go beyond the word timing in Stutter's calculations by adding the idea of a text fragment into the data. Each word now knows which fragment it is a part of, and Stutter uses these text-fragments to apply <mark> tags around the relevant text on screen. This will not be 100% perfect because we're running our site through the Readability API for full-page processing and the text Stutter gets back may not match the DOM perfectly. Still, even a close match gives a usable progress feeling. This non-match is noticable in the video when the text first starts the second paragraph.

I was worried this was going to massively slow down processing times, but it doesn't seem that bad. I'll have to stress test on much larger pages still. This *does* do DOM traversal and manipulation, which isn't great, but there's no way around that.

Next up I need to investigate why the contextual Stuttering has issues (later part of video). Stuttering a selection isn't proccesing properly and can hang.

I also uncovered a bug related to localization that I'll file separately. I'm using localized punctuation for splitting badly processed words, but I've got latin punctuation hard-coded in the speed calc code in `word.js`. That needs some love.
